src/Microsoft.DotNet.Build.CloudTestTasks/UploadClient.cs

```diff
@@ -208,6 +210,57 @@ public string EncodeBlockIds(int numberOfBlocks, int lengthOfId)
                 }
             }
         }
+
+        public async Task<bool> FileEqualsExistingBlobAsync(
+            string accountName,
+            string accountKey,
+            string containerName,
+            string filePath,
+            string destinationBlob,
+            int uploadTimeout)
+        {
+            using (var client = new HttpClient
+            {
+                Timeout = TimeSpan.FromMinutes(uploadTimeout)
+            })
+            {
+                log.LogMessage(
+                    MessageImportance.Low,
+                    $"Downloading blob {destinationBlob} to check if identical.");
+
+                string urlGetBlob = AzureHelper.GetBlobRestUrl(accountName, containerName, destinationBlob);
```

Comment:
nit: I don't like the variable name because if I just read it somewhere I would assume that `urlGetBlob` represented a Url type.  Alternatively, putting "Url" as a suffix does not have the same connotation.  Maybe just 'blobUrl'.